source/Cosmos.System2/FileSystem/Disk.cs

```diff
@@ -0,0 +1,296 @@
+            mbrData[location + 2] = 0; //Starting sector
+            mbrData[location + 3] = 0x0; //Starting Cylinder
+            mbrData[location + 4] = 83;//normal partition
+            mbrData[location + 5] = 0xFE; //ending head
+            mbrData[location + 6] = 0x3F; //Ending Sector
+            mbrData[location + 7] = 0x40;//Ending Cylinder
+
+            //Starting Sector
+            byte[] startingSectorBytes = BitConverter.GetBytes(startingSector);
+            mbrData[location + 8] = startingSectorBytes[0];
+            mbrData[location + 9] = startingSectorBytes[1];
+            mbrData[location + 10] = startingSectorBytes[2];
+            mbrData[location + 11] = startingSectorBytes[3];
+
+            //Total Sectors in partition
+            byte[] total = BitConverter.GetBytes(amountOfSectors);
+            mbrData[location + 12] = total[0];
+            mbrData[location + 13] = total[1];
+            mbrData[location + 14] = total[2];
+            mbrData[location + 15] = total[3];
+
+            //Boot flag
+            byte[] boot = BitConverter.GetBytes((ushort)0xAA55);
+            mbrData[510] = boot[0];
+            mbrData[511] = boot[1];
+
+            //Save the data
+            Host.WriteBlock(0, 1, ref mbrData);
+        }
+        /// <summary>
+        /// Deletes a partition
+        /// </summary>
+        /// <param name="index">Partition index starting from 0</param>
+        public void DeletePartition(int index)
+        {
+            if (GPT.IsGPTPartition(Host))
+            {
+                throw new Exception("Deleting partitions with GPT style not yet supported!");
+            }
+            var location = 446 + 16 * index;
+
+            byte[] mbr = Host.NewBlockArray(1);
+            Host.ReadBlock(0, 1, ref mbr);
+            for (int i = location; i < location + 16; i++)
+            {
+                mbr[i] = 0;
+            }
+            Host.WriteBlock(0, 1, ref mbr);
+        }
+        /// <summary>
+        /// Deletes all partitions on the disk.
+        /// </summary>
+        public void Clear()
+        {
+            if (GPT.IsGPTPartition(Host))
+            {
+                throw new Exception("Removing all partitions with GPT style not yet supported!");
+            }
+        }
+
+        public void FormatPartition(int index, string format, bool quick = true)
+        {
+            var part = Partitions[index];
+
+            var xSize = (long)(Host.BlockCount * Host.BlockSize / 1024 / 1024);
+
+            if (format.StartsWith("FAT"))
+            {
+                FatFileSystem.CreateFatFileSystem(part.Host, VFSManager.GetNextFilesystemLetter() + ":\\", xSize, format);
+                Mount();
+            }
+            else
+            {
+                throw new NotImplementedException(format + " formatting not supported.");
+            }
+        }
+
+        private FileSystem[] MountedPartitions = new FileSystem[4];
```

Comment:
Why does it have length 4?

Reply:
Because there can only be 4 partitions in mbr partitioning scheme.

Reply:
Ok thanks